pkg/agent/datamodel/types.go

```diff
@@ -350,6 +350,19 @@ const (
 	Webhook AuthenticatorType = "webhook"
 )
 
+type KuberentesAuthMethod string
+
+const (
+	UnknownAuthMethod         KuberentesAuthMethod = ""
+	UseArcMsiToMakeCSR        KuberentesAuthMethod = "UseArcMsiToMakeCSR"
+	UseAzureMsiToMakeCSR      KuberentesAuthMethod = "UseAzureMsiToMakeCSR"
+	UseArcMsiDirectly         KuberentesAuthMethod = "UseArcMsiDirectly"
+	UseAzureMsiDirectly       KuberentesAuthMethod = "UseAzureMsiDirectly"
+	UseSecureTLSBootstrapping KuberentesAuthMethod = "UseSecureTLSBootstrapping"
+	//nolint:gosec // this is a const string to use in switch statements, not hardcoded credentials
+	UseTLSBootstrapToken KuberentesAuthMethod = "UseTLSBootstrapToken"
+)
```

Comment:
Do you want to implement it only for aks-node-controller or for old-way as well?